### PR TITLE
[chore][exporter/coralogixexporter] Enable goleak check

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

This enables the `goleak` check on all tests in this package, to ensure the exporter isn't leaking goroutines. The only updates here are tests, as some were missing necessary `shutdown` calls. I also updated some variable naming to match types, such as `le` for  a logs exporter instead of `me`.

**Link to tracking Issue:** <Issue number if applicable>
#30438

**Testing:** <Describe what testing was performed and which tests were added.>
All tests are passing, including added `goleak` check.

### CODE REVIEW

**Comment by crobert-1** [comment]
Filed #30805 for failing test, unrelated to this change.

**Review by atoulme — APPROVED** [review]
changes make sense to me, approved. A codeowner should review as well.

**Review by povilasv — APPROVED** [review]
Thanks for this.

**Review by povilasv — APPROVED** [review]

File: exporter/coralogixexporter/package_test.go
Comment (codeboten) [inline exporter/coralogixexporter/package_test.go:16]: Is the ignore here needed? I removed a bunch of these after the change to remove much of the opencensus dependencies was merged in core.

Comment (crobert-1) [reply exporter/coralogixexporter/package_test.go]: Thanks for catching this @codeboten, you're right, this is no longer needed. I've removed it.

**Review by codeboten — APPROVED** [review]

**Review by povilasv — APPROVED** [review]

**Pull request merged**
